controlplane/src/core/bufservices/PlatformService.ts

```diff
@@ -203,8 +223,24 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
 
       return handleError<PlainMessage<CreateFederatedSubgraphResponse>>(logger, async () => {
         const authContext = await opts.authenticator.authenticate(ctx.requestHeader);
-
         const subgraphRepo = new SubgraphRepository(opts.db, authContext.organizationId);
+        const userRepo = new UserRepository(opts.db);
+
+        const hasAccess = await userRepo.checkUserAccess({
```

Comment:
I think we should add this to the `authenticate` and expose it through `writeAccess` and `isAdmin` property. That removes boilerplate.

Reply:
Good idea. Done